### PR TITLE
Bugfix/#1 assignment changes

### LINKED ISSUE #1: Sensor/Actuator assignment changes after update
After calling update for the first time, the sensor object is assigned to another sensor on the xs1. However, it seems that this only happens, after update() is called for the first time on the update. After that is stays stable. 

```
>>> from xs1_api_client import api as xs1api
>>> from xs1_api_client import api_constants

>>> api = xs1api.XS1('192.168.xxx.xxx')
>>> sensors = api.get_all_sensors()
>>> print(sensors[0])
Sensor: Wohn_Temp2 (ID: 7, Enabled: True, Type: ActuatorType.TEMPERATURE, Value: 22.1, New Value: None, Unit: °C)
>>> sensors[0].update()
>>> print(sensors[0])
Sensor: Gaeste_O_Temp2 (ID: 7, Enabled: True, Type: ActuatorType.TEMPERATURE, Value: 0.0, New Value: None, Unit: °C)
```

### PR DESCRIPTION
**Pull request by markusressel**
fixes #1 
This PR separates the ID and Number properties inside `XS1Device` to prevent changes in the order of the Web GUI to affect the order devices are updated.

The `id` property is a unique identifier that does **not** change even when swapping devices in the XS1 Web GUI using the "Austauschen" button.
The `number` property however **does** change when the position of a device is swapped with another one, so it can not be used for identification but only for ordering.

Note: for this change to propagate to Home Assistant a new release has to be incorporated.

### CODE REVIEW

**Comment by markusressel** [comment]
@fti7 are you able to quickly test if this fix also works inside home assistant?

**Comment by fti7** [comment]
@markusressel Unfortunately it seems that it doesnt fix the issue :-(
The "ID" isnt known without update() first. Did you forgot to commit something? looks incomplete
Thanks for looking into it :-)
```
Sensor 1: Wohnzimmer_Temp (21.3)
Sensor 2: Wohnzimmer_Hygro (53.0)
Sensor 3: Schlafzimmer_Temp (22.3)
Sensor 4: Schlafzimmer_Hygro (49.0)
Sensor 7: Bad_Temp (22.5)
Sensor 8: Bad_Hygro (46.8)
Sensor 9: Aussen_Temp (17.5)
Sensor 10: Aussen_Hygro (62.5)
xxxxx
Sensor 1: Wohnzimmer_Temp (21.3)
Sensor 2: Wohnzimmer_Hygro (53.0)
Sensor 3: Schlafzimmer_Temp (22.3)
Sensor 4: Schlafzimmer_Hygro (49.0)
Sensor 7: Sensor_15 (0.0)
Sensor 8: Sensor_16 (0.0)
Sensor 9: Bad_Temp (22.5)
Sensor 10: Bad_Hygro (46.8)
```

**Comment by markusressel** [comment]
It is the other way around. ID is available after `get_all_sensors()`, while the number is only available after an `update()` call. However, it is implicitly available from the list index when retrieving all sensors.

After some investigation I found the issue:
The XS1 api (for some strange reason unknown to me) always refers to devices via its **number** when using the REST api and **does not** allow to use the **id** instead (I tried to use an `id` param, but it didn't work). When `id != number` this causes the observed behavior of a device magically changing its device association after an `update()` call.

You were totally right to derive the `number` from the list location when retrieving the list of sensors and actuators. I added (a slightly altered version of) this code back in.

Since the API uses the number instead of the ID, **there is still one thing to note**:
**xs1-api-client** was designed to have reusable device objects, meaning there is no need to recreate a device object but be able to update it. When a user changes the order of devices within the XS1 gateway the id does not change but the number does, however, **we do not know about this change**. This means there is no way for us to tell that the device number we use for an already created device object does not correspond to the correct device (the same id) anymore without fetching all devices again, which requires the recreation of all device objects. Maybe it would be possible to find out where the device was moved based on its configuration, but since this would require a lot of work for a small edge case I do not think it is worth the effort.

TL;DR: Do not change the order of the devices in the XS1 Gateway, and if you do, restart the service using **xs1-api-client** to force a re-fetch of all devices.

I will add a notice about this to the README.

**Comment by fti7** [comment]
Awesome. Yes the XS1 API Sucks :-(
I also dont think its worth the effort... not sure how many XS1 Devices are still arround in 2020

Tried it now in HASS and it works :-)
Looking forward for an Release/HASS Upstream Change

**Comment by markusressel** [comment]
See https://github.com/home-assistant/core/pull/36500
